src/ts/contentScripts/live-preview/index.tsx

```diff
@@ -0,0 +1,267 @@
+        this.addIframeToBody()
+        this.scrollToTopHack()
+        this.attachMouseListeners()
+    }
+
+    private addIframeToBody() {
+        document.body.appendChild(this.iframe)
+    }
+
+    private attachMouseListeners() {
+        document.addEventListener('mouseover', this.mouseOverListener)
+        document.addEventListener('mouseout', this.mouseOutListener)
+    }
+    private destoryMouseListeners() {
+        document.removeEventListener('mouseover', this.mouseOverListener)
+        document.removeEventListener('mouseout', this.mouseOutListener)
+    }
+
+    private mouseOverListener = (e: Event) => {
+        const target = e.target as HTMLElement
+        const isPageRef = this.isTargetPageRef(target)
+        if (isPageRef) {
+            const text = this.getTargetInnerText(target)
+            this.hoveredElement = target
+            const url = Navigation.getPageUrlByName(text)
+            if (url) {
+                this.prepIframeForDisplay(url)
+                this.setTimerForPopup(target)
+            }
+        }
+    }
+
+    private mouseOutListener = (e: MouseEvent) => {
+        const nextTarget = e.relatedTarget as HTMLElement
+        if (this.shouldRemoveOnMouseOut(nextTarget)) {
+            this.hidePreview()
+        }
+    }
+
+    private hidePreview() {
+        this.hoveredElement = null
+        this.clearPopupTimeout()
+        this.resetIframeForNextHover()
+        this.destroyPopper()
+    }
+
+    private shouldRemoveOnMouseOut(nextTarget: HTMLElement) {
+        const isNotHoveringOverTarget = nextTarget !== this.hoveredElement
+        const isNotHoveringOverIframe = nextTarget !== this.iframe
+        return isNotHoveringOverIframe && isNotHoveringOverTarget
+    }
+
+    private setTimerForPopup(target: HTMLElement) {
+        if (!this.popupTimeout) {
+            this.popupTimeout = window.setTimeout(() => {
+                this.showPreview()
+                this.makePopper(target)
+            }, this.popupTimeoutDuration)
+        }
+    }
+
+    private getTargetInnerText(target: HTMLElement) {
+        // remove '#' for page tags
+        const isPageRefTag = this.isTargetPageRefTag(target)
+        return isPageRefTag ? target.innerText.slice(1) : target.innerText
+    }
+
+    private isTargetPageRefTag(target: HTMLElement) {
+        return target.classList.contains('rm-page-ref-tag')
+    }
+
+    private isTargetPageRef(target: HTMLElement) {
+        return target.classList.contains('rm-page-ref')
+    }
+
+    private destroyPopper() {
+        if (this.popper) {
+            this.popper.destroy()
+            this.popper = null
+        }
+    }
+
+    private resetIframeForNextHover() {
+        this.scrollToTopOnMouseOut()
+        this.iframe.style.pointerEvents = 'none'
+        this.iframe.style.opacity = '0'
+        this.iframe.style.height = '0'
+        this.iframe.style.width = '0'
+    }
+
+    private scrollToTopOnMouseOut() {
```

Comment:
nit: can you place the scroll to top hacks close to each other and extract commonality between them?